### DISCUSSION TITLE
How to migrate run history from a repository to Definitions

### DISCUSSION THREAD

**Discussion by dpeng817** [dagster-io/dagster#14049] [Q&A] [answered]
Consider switching a repository named `foo_repo` in code location `bar_location` to a `Definitions` object in "baz_location". Let's say that the workspace file is at `path_to_file`
Run the following command:
```shell
dagster run migrate-repository --from "bar_location@foo_repo" --workspace "path_to_file/workspace.yaml" --location "baz_location"
```
The command will prompt you to type `MIGRATE` to confirm that you would like to execute the migration.

**Note** there are additional ways to target the definitions object that you are moving history to. Consider for example, a definitions object living in a python module `baz`. 
```shell
dagster run migrate-repository --from "bar_location@foo_repo" --module "baz" --repository "__repository__"
```
Since, under the hood, we create a repository called `__repository__` for every `Definitions` object.

**Comment by dpeng817** [comment, marked as answer]
answer in description

**Reply by jimenezj8** [reply]
@dpeng817 sorry to dredge this up after so many months, this is just the only documentation I've found for this particular problem.

After performing the migration, it seems that the run history is still not populating for the job configured with `Definitions`. I noticed that the run history was actually preserved in the workspace (under the job configured with `@repository`) despite the job in the migrated code location not showing with any run history. Any clues as to where I might be going wrong?

**Comment by NicolasPA** [comment]
Worked great after a few tries because the parameters are not that obvious, thanks!

People should note that if there's more than one job, they have to run the command for every single job, which is kind of a hassle.
It would be nice if the default behavior would be to migrate all jobs (after asking for permission as it already does), alternatively providing a symbol such as `'*'` to select all jobs. 
Given the error when I didn't select a job, it already has the code to fetch the existing jobs : `Error: Must provide --job as there is more than one job in repository_object_name. Options are: ['job', 'job1', 'job2', 'job3'].`.

Here's another example that worked for us, our use case was a change in the path (Python package and module name) of where the repository object is stored :

```shell
dagster run migrate-repository
	# old location of the repository object 
	--from "repository_object_name@python_package1.python_package2.old_python_package3.old_python_module_name:repository_object_name" 
	# name of the job to migrate
	--job "job1" 
	# current workspace address referencing the repository
	--workspace "python_package1/python_package2/new_python_package3/workspace.yaml" 
	# new repository object location
	--location "python_package1.python_package2.new_python_package3.new_python_module_name:repository_object_name"
```

Related point: we also lost the history in our schedules and sensor, and it seems there's no command to migrate those.

Ideally, it would be nice to have a single command to tell Dagster that a repository location has been changed and Dagster would migrate everything, so it looks as if the repository location hadn't changed with no loss of history at all.
